redux/selectors/projects.js

```diff
@@ -1,21 +1,36 @@
 import _ from 'lodash'
 import { createSelector } from 'reselect-change-memoize'
 
+const filterProjectsByQuery = (projects, projectQuery) => {
+  const lowerCaseProjectQuery = _.toLower(projectQuery)
+
+  return _.filter(projects, project => {
+    const lowerCaseProjectName = _.toLower(project.name)
+    return _.includes(lowerCaseProjectName, lowerCaseProjectQuery)
+  })
+}
+
+const filterRecentlyFailedProjects = projects => {
+  return _.filter(projects, project => {
+    return !_.chain(project.branches)
+      .filter(branch => branch.latestBuild.status === 'failed')
+      .isEmpty()
+      .value()
+  })
+}
+
 export default createSelector(
   'projects',
   state => state.projects,
-  projectsState => {
-    const lowerCaseProjectQuery = _.toLower(projectsState.projectQuery)
-
-    // We would normally do this server side, but this allows us to give a selector example
-    const filteredProjects = _.filter(projectsState.projects, project => {
-      const lowerCaseProjectName = _.toLower(project.name)
-      return _.includes(lowerCaseProjectName, lowerCaseProjectQuery)
-    })
+  state => state.filters,
+  (projectsState, filtersState) => {
+    // We would normally do this server side, but this allows us to give a nice selector example
+    const projectsFilteredByQuery = filterProjectsByQuery(projectsState.projects, filtersState.projectQuery)
+    const recentlyFailedProjects = filterRecentlyFailedProjects(projectsFilteredByQuery)
 
     return {
       ...projectsState,
-      projects: filteredProjects,
+      projects: filtersState.showFailed ? recentlyFailedProjects : projectsFilteredByQuery,
```

Comment:
does this mean you can use both filters simulataneously?

Reply:
Yeah, `filterRecentlyFailedProjects` takes `projectsFilteredByQuery` as it's arg

Reply:
how to code it is a seperate issue.

But I'd like to be able to check the box and search 'wash' and only see failures related to wash doctors.

Reply:
This does this already